File: c#/Project/Dal_Repository/models/Product.cs
using System;
using System.Collections.Generic;

namespace Dal_Repository.models;

public partial class Product
{
    public string? ProductName { get; set; }

    public int ProductId { get; set; }

    public int? TargetAge { get; set; }

    public int? CategoryId { get; set; }

    public int? CompanyId { get; set; }

    public string? Picture { get; set; }

    public string? ProductDes { get; set; }

    public int? Price { get; set; }

    public DateOnly? LastUpdate { get; set; }

    public virtual Category? Category { get; set; }

    public virtual Company? Company { get; set; }

    public virtual ICollection<PurchaseDetail> PurchaseDetails { get; set; } = new List<PurchaseDetail>();
}

File: Angular/src/app/models/product.ts
export class Product {
  id: number;
  productName: string;
  categoryName: number;
  companyName: number;
  productDes: string;
  price: number;
  targetAge: number;
  picture: string;
  lastUpdate: Date;
  amount: number;
  constructor(id: number, productName: string, categoryName: number, companyName: number, productDes: string, price: number, lastUpdate: Date, targetAge: number, picture: string) {
    this.id = id;
    this.productName = productName;
    this.companyName = companyName;
    this.categoryName = categoryName;
    this.productDes = productDes;
    this.lastUpdate = lastUpdate;
    this.picture = picture;
    this.price = price;
    this.targetAge = targetAge;
    this.amount = 1
  }
}
